log/include/gz/utils/log/Logger.hh

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2024 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#ifndef GZ_UTILS_LOG_LOGGER_HH_
+#define GZ_UTILS_LOG_LOGGER_HH_
+
+#include <spdlog/spdlog.h>
+#include <memory>
+#include <string>
+#include <gz/utils/ImplPtr.hh>
+
+namespace gz::utils::log
+{
+  /// \brief Gazebo console and file logging class.
+  /// This will configure spdlog with a sane set of defaults for logging to the
+  /// console as well as a file.
+  class GZ_UTILS_VISIBLE Logger
```

Comment:
Since this is a component, the macro should be `GZ_UTILS_LOG_VISIBLE`, I think.

Reply:
Done in [be88966](https://github.com/gazebosim/gz-utils/pull/134/commits/be88966ecb3a8721073cbf203e86362ef4209777).